rpc/handler.go

```diff
@@ -468,16 +472,29 @@ func (h *handler) handleCallMsg(ctx *callProc, msg *jsonrpcMessage) *jsonrpcMess
 
 	case msg.isCall():
 		resp := h.handleCall(ctx, msg)
-		var ctx []interface{}
-		ctx = append(ctx, "reqid", idForLog{msg.ID}, "duration", time.Since(start))
+		var logCtx []interface{}
+		logCtx = append(logCtx, "reqid", idForLog{msg.ID}, "duration", time.Since(start))
 		if resp.Error != nil {
-			ctx = append(ctx, "err", resp.Error.Message)
+			logCtx = append(logCtx, "err", resp.Error.Message)
 			if resp.Error.Data != nil {
-				ctx = append(ctx, "errdata", resp.Error.Data)
+				// If the log level is debug, log the full error data. Otherwise, try to truncate it.
```

Comment:
Is this the right value to branch on ? Do rpc runners not run nodes with log level >= debug? I feel like we should either unilaterally truncate the error data in the log or have a separate flag to control error truncation.

I lean towards unilaterally truncating.

Reply:
I think we should let debug level logger to show full logs without truncating to actually see the full error (probably in case after seeing truncated logs). 

I was under impression rpc nodes would run with Info/Warn (not debug).

Reply:
As an RPC node, it feels unexpected that it's "safe" for me to expose my endpoint if my log level is below debug, but once I hit debug it is not longer "safe".

Reply:
true, and that branch truncates logs if the log level > debug (warn, info) which is safe (if I'm not missing something else). `log.Enabled(context.Background(), log.LvlDebug)` would return `true` if the logger is operating at `debug` or below (`trace`). otherwise we will truncate it (if info/warn).  

It might be a bit strange to see different outputs (non-truncated) for a `Warn` log when you switch to debug, but I can't think a better way to present the full non-truncated err data. Alternatively maybe we can print both a warn message with truncated data and then a debug message with non-truncated data.

Reply:
I didn't even know we printed the error data in full in the first place. It seems to have been added in https://github.com/ethereum/go-ethereum/pull/21083. Really don't think it's necessary to preserve the full data in logs, since they are just for debugging failed calls. So let's just always truncate the data.

Reply:
removed the debug lvl condition